tests/test_workdir.py

```diff
@@ -57,12 +60,98 @@ def test_directory_structure_generated(self, spatial_directory_list):
             assert path.exists()
 
 
-class TestCleanDirectoryTree:
-    """Tests for `clean_directory_tree()`."""
+class TestCleanFiles:
+    """Tests for `clean_realisation_files()` and `clean_submission_files()`."""
 
-    @pytest.mark.parametrize("test_path", [Path("runs"), Path("src")])
-    def test_clean_directory_tree(self, test_path):
-        """Success case: directory tree does not exist after clean."""
-        test_path.mkdir()
-        clean_directory_tree()
-        assert not test_path.exists()
+    # Reset internal.CWD to suit pytest testing infrastructure
+    @pytest.fixture(autouse=True)
+    def _set_internal_cwd(self, monkeypatch):
+        """Sets internal.CWD to pytest's working directory."""
+        monkeypatch.setattr(internal, "CWD", Path.cwd())
+
+    # Helper functions
+    def _create_files_in_cwd(self, filenames: List[str]):
+        """Given a list of filenames, create files in current working directory."""
+        for filename in filenames:
+            filename_path = internal.CWD / filename
+            filename_path.touch()
+
+    def _check_if_any_files_exist(self, filenames: List[str]):
+        """Given a list of filenames, check if any of them exist w.r.t. current working directory."""
+        return any((internal.CWD / filename).exists() for filename in filenames)
+
+    @pytest.fixture()
+    def src_path(self) -> Path:
+        """Mock internal.SRC_DIR."""
+        src_path = internal.CWD / Path("src")
+        src_path.mkdir()
+        return src_path
+
+    @pytest.fixture()
+    def runs_path(self) -> Path:
+        """Mock internal.RUN_DIR."""
+        runs_path = internal.CWD / Path("runs")
+        runs_path.mkdir()
+        return runs_path
+
+    @pytest.fixture()
+    def revision_log_files(self) -> List[Path]:
+        """Create sample files of the form rev_number-*.log."""
+        rev_log_files = ["rev_number-0.log", "rev_number-200.log"]
+        self._create_files_in_cwd(rev_log_files)
+        return rev_log_files
+
+    @pytest.fixture()
+    def pbs_job_files(self) -> List[Path]:
+        """Create sample files of the form benchmark_cable_qsub.sh*."""
+        pbs_job_files = ["benchmark_cable_qsub.sh.o21871", "benchmark_cable_qsub.sh"]
+        self._create_files_in_cwd(pbs_job_files)
+        return pbs_job_files
```

Comment:
The typing is not coherent with the code. Currently `pbs_job_files` is list(str) but it should be list(path).

```suggestion
    def pbs_job_files(self) -> List[Path]:
        """Create sample files of the form benchmark_cable_qsub.sh*."""
        pbs_job_files = [Path("benchmark_cable_qsub.sh.o21871"), Path("benchmark_cable_qsub.sh")]
        self._create_files_in_cwd(pbs_job_files)
        return pbs_job_files
```